tool/format_ignores.py

```diff
@@ -0,0 +1,226 @@
+# /// script
+# requires-python = ">=3.10"
+# dependencies = [
+#     "numtype[numpy]",
+# ]
+#
+# [tool.uv.sources]
+# numtype = {path = ".."}
+# ///
+
+"""Format pyright ignore comments with proper sorting and spacing.
+
+Ensures that pyright ignore comments in Python files have their entries sorted
+alphabetically and formatted with proper spacing between items.
+
+Usage: uv run tool/format_ignores.py [-h] [--extensions EXTENSIONS] [--check] [PATH]
+"""
+
+import argparse
+import os
+import re
+import sys
+from collections.abc import Iterable, Sequence
+from pathlib import Path
+
+
+def _sort_ignore_list(ignore_text: str, /) -> str:
+    """
+    Sort the items in a pyright ignore list and ensure proper spacing.
+
+    Parameters
+    ----------
+    ignore_text : str
+        Text inside the ignore brackets
+
+    Returns
+    -------
+    str
+        Sorted and formatted ignore list
+    """
+    # Extract items from the ignore list
+    items = ignore_text.strip("[]").split(",")
+
+    # Clean up each item and sort
+    cleaned_items = [item.strip() for item in items if item.strip()]
+    sorted_items = sorted(cleaned_items)
+
+    # Join with comma and space
+    return ", ".join(sorted_items)
+
+
+def _process_file(
+    file_path: str | os.PathLike[str],
+    /,
+    *,
+    check_only: bool = False,
+) -> bool:
+    """
+    Process a file to sort pyright ignore comments.
+
+    Parameters
+    ----------
+    file_path : str or os.PathLike[str]
+        Path to the file to process
+    check_only : bool, default=False
+        If True, don't modify the file, just check if changes are needed
+
+    Returns
+    -------
+    bool
+        True if the file was modified or would be modified (in check mode)
+    """
+    path_obj = Path(file_path)
+    content = path_obj.read_text(encoding="utf-8")
+
+    # Find all pyright ignore comments and sort them
+    pattern = r"(# pyright: ignore\[)([^\]]+)(\])"
+
+    def replace_match(match: re.Match[str], /) -> str:
+        prefix = match.group(1)
+        ignore_list = match.group(2)
+        suffix = match.group(3)
+        sorted_ignore = _sort_ignore_list(ignore_list)
+        return f"{prefix}{sorted_ignore}{suffix}"
```

Comment:
How about we also include mypy's `# type: ignore` comments here?

By also taking any weird whitespace into account, it could look something like

```suggestion
    # Find all ignore comments and sort them
    pattern = r"\s*#\s*(pyright|type):\s*ignore\[([a-zA-Z\-,\s]+)\]"

    def replace_match(match: re.Match[str], /) -> str:
        checker = match.group(1)
        sorted_ignores = _sort_ignore_list(match.group(2))
        return f"  # {checker}: ignore[{sorted_ignores}]"
```

Reply:
Now that I think about it, it's probably better to use `re.compile` here. Otherwise the regex has to be recompiled for each file it encounters.